filter_main.go

```diff
@@ -1,6 +1,9 @@
 //go:generate ragel -Z lexer.rl
-//go:generate -command yacc go tool yacc
-//go:generate yacc -o parser.go -p filter parser.y
+//go:generate goyacc -o parser.go -p filter parser.y
+// HACK: we cant modify yacc directly to pass the state
```

Comment:
Could you pls explain your idea?
What about Windows platform?

Reply:
The issue is that `checkFieldNameVsTypeValue` in `parser.go` [requires access to the Context](https://github.com/kor44/gofilter/blob/e1cc660f6226db176fa6a3a4cc0ae1abc2ba8723/parser.go#L31).

But the `filterLexer` generated by yacc hard-codes [the interface](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.9.1:cmd/goyacc/yacc.go;l=3232), so we need to modify the `parser.go` to add the required Context.

As said in the comment, this is a hacky way to get it to work and I'm open to suggestions.
Unless you are working on gofilter itself, you won't ever need to recompile the parser because the `parser.go` file is shipped pre-compiled.

Reply:
May be change structure filterLexerImpl?

Reply:
Also [hardcoded](https://cs.opensource.google/go/x/tools/+/refs/tags/v0.9.1:cmd/goyacc/yacc.go;l=3242)